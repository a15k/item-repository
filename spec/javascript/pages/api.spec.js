import renderer from 'react-test-renderer';
import Api from 'pages/api';

jest.mock('pages/users/delete-placeholder', () => {
  return () => <div className="delete-placeholder" />;
});

jest.mock('pages/users/disabled-power-user-toggle', () => {
  return () => <input type="checkbox" className="disabled-power-user" readOnly checked />;
});

describe(Api, () => {
  let props = {};

  it('renders combinded actions', () => {
    const api = renderer.create(<Api {...props} />, {
      createNodeMock: e => {
        return document.createElement(e.type);
      },
    });
    expect(api.toJSON()).toMatchSnapshot();
    api.unmount();
  });

});
