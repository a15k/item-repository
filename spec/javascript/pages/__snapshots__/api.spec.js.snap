// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api renders combinded actions 1`] = `
<div>
  <div
    className="access-tokens"
  >
    <div
      className="sc-bxivhb cjKfDO"
    >
      <h3>
        Access Tokens
      </h3>
      <button
        className="sc-bdVaJa gZEDXH btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 icon"
          data-icon="plus"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span>
          Add
        </span>
      </button>
    </div>
    <p>
      Access tokens are used to access the A15K api
    </p>
    <div
      onScroll={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflowX": "hidden",
        }
      }
    >
      <div
        className="react-swipeable-view-container"
        style={
          Object {
            "WebkitFlexDirection": "row",
            "WebkitTransform": "translate(0%, 0)",
            "WebkitTransition": "all 0s ease 0s",
            "direction": "ltr",
            "display": "flex",
            "flexDirection": "row",
            "height": null,
            "transform": "translate(0%, 0)",
            "transition": "all 0s ease 0s",
            "willChange": "transform",
          }
        }
      >
        <div
          aria-hidden={false}
          className={undefined}
          data-swipeable="true"
          style={
            Object {
              "WebkitFlexShrink": 0,
              "flexShrink": 0,
              "overflow": "auto",
              "width": "100%",
            }
          }
        >
          <ul
            className="list-group"
          />
        </div>
        <div
          aria-hidden={true}
          className={undefined}
          data-swipeable="true"
          style={
            Object {
              "WebkitFlexShrink": 0,
              "flexShrink": 0,
              "overflow": "auto",
              "width": "100%",
            }
          }
        >
          <form
            className="sc-ifAKCX gitGbc"
          >
            <div
              className="form-group"
            >
              <label
                className=""
                htmlFor="name"
              >
                Name
              </label>
              <input
                className="form-control"
                defaultValue=""
                id="name"
                name="name"
                type="text"
              />
            </div>
            <div
              className="sc-bwzfXH ehHYlP"
            >
              <button
                className="sc-bdVaJa gZEDXH btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-chevron-left fa-w-10 icon"
                  data-icon="chevron-left"
                  data-prefix="fas"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span>
                  Cancel
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="sc-EHOje chiHEN"
  />
  <div
    className="interactions"
  >
    <div
      className="sc-bxivhb cjKfDO"
    >
      <h3>
        Interactions
      </h3>
      <button
        className="sc-bdVaJa gZEDXH btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 icon"
          data-icon="plus"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span>
          Add
        </span>
      </button>
    </div>
    <p>
      The interactions api is used to flag and add other metadata to assessments
    </p>
    <div
      onScroll={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflowX": "hidden",
        }
      }
    >
      <div
        className="react-swipeable-view-container"
        style={
          Object {
            "WebkitFlexDirection": "row",
            "WebkitTransform": "translate(0%, 0)",
            "WebkitTransition": "all 0s ease 0s",
            "direction": "ltr",
            "display": "flex",
            "flexDirection": "row",
            "height": null,
            "transform": "translate(0%, 0)",
            "transition": "all 0s ease 0s",
            "willChange": "transform",
          }
        }
      >
        <div
          aria-hidden={false}
          className={undefined}
          data-swipeable="true"
          style={
            Object {
              "WebkitFlexShrink": 0,
              "flexShrink": 0,
              "overflow": "auto",
              "width": "100%",
            }
          }
        >
          <ul
            className="list-group"
          />
        </div>
        <div
          aria-hidden={true}
          className={undefined}
          data-swipeable="true"
          style={
            Object {
              "WebkitFlexShrink": 0,
              "flexShrink": 0,
              "overflow": "auto",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
